Sources/Knit/Module/ModuleAssembly.swift

```diff
@@ -40,6 +45,24 @@ public extension ModuleAssembly {
             return self.resolverType == $0.resolverType
         }
     }
+
+    static var _assemblyFlags: [ModuleAssemblyFlags] {
+        var result: [ModuleAssemblyFlags] = []
+        if self is any AutoInitModuleAssembly.Type {
+            result.append(.autoInit)
+        }
+        if self is any AbstractAssembly.Type {
+            result.append(.abstract)
+        }
+        return result
+    }
+
+    static func _autoInstantiate() -> (any ModuleAssembly)? {
+        if let autoInit = self as? any AutoInitModuleAssembly.Type {
+            return autoInit.init()
+        }
+        return nil
```

Comment:
```suggestion
    static func _autoInstantiate() -> (any ModuleAssembly)? {
        return nil
```
Should we remove the `as?` check here and always return `nil`?

Reply:
Or alternately remove the default extension implementation entirely?

Reply:
If I take out the default implementation then I'll have to add it manually for any test cases where it's not being autogenerated.